tools/clang/unittests/HLSLExec/ExecutionTest.cpp

```diff
@@ -11096,6 +11135,402 @@ TEST_F(ExecutionTest, PackUnpackTest) {
+          if (diff > tolerance) {
+            vectorsMatch = false;
+            mismatchedIndexes.push_back(i);
+          }
+        }
+    }
+
+    // Print the mismatched indexes and their corresponding elements
+    if (!mismatchedIndexes.empty()) {
+        for (size_t index : mismatchedIndexes) {
+          std::wstringstream wss(L"");
+          wss << L"Mismatch at Index: " << index;
+          wss << L" Expected Value:" << vec1[index];
+          wss << L" Actual Value:" << vec2[index];
+          WEX::Logging::Log::Error(wss.str().c_str());
+        }
+    }
+
+    return vectorsMatch;
+}
+
+template <typename T, typename = std::enable_if_t<std::is_arithmetic_v<T>>>
+T parseStringToNumber(std::wstring& number) {
+  if constexpr (std::is_same_v<T, int>) {
+    return std::stoi(number);
+  } else if constexpr (std::is_same_v<T, float>) {
+    return std::stof(number);
+  } else if constexpr (std::is_same_v<T, double>) {
+    return std::stod(number);
+  }
+  else
+  {
+    VERIFY_IS_TRUE(false, L"Unsupported type for parsing string to number.");
+    return 0;
+  }
+}
+
+// Parse a vector of WEX::Common::Strings to a vector of numbers.
+// Expected string format is a comma separated list of numbers.
+// Strings with no commas could also be used for single numbers.
+// For example:
+// Three strings: ["1.0,2.0,3.0,4.0", "5.0,6.0,7.0,8.0", "9.0,10.0,11.0,12.0"]
+// would parse to 3 vectors of 4 elements each.
+// Or three strings ["1" , "2", "3"] will parse to 3 single elements vectors.
+// Vectors for a single element is a little weird, but it allows this helper
+// function to be very flexible.
+template <typename T, typename = std::enable_if_t<std::is_arithmetic_v<T>>>
+std::vector<std::vector<T>> parseStringsToNumbers(const std::vector<WEX::Common::String>* input) {
+  std::vector<std::vector<T>> result = {};
+  if (input == nullptr || input->empty()) {
+      return result;
+  }
+
+  for (const auto& wexStr : *input) {
+
+    std::wstring wStr(wexStr);
+    std::wstringstream wss(wStr);
+    std::wstring number;
+    std::vector<T> vec;
+
+    while (std::getline(wss, number, L',')) {
+        vec.push_back(parseStringToNumber<T>(number));
+    }
+    result.push_back(vec);
+  }
+
+  // Sanity check for bad test data.
+  VERIFY_IS_TRUE(result.size() == input->size(), L"Result vector size does not match input vector size.");
+
+  return result;
+}
+
+TEST_F(ExecutionTest, LongVector_BinaryOpTest) {
+  WEX::TestExecution::SetVerifyOutput verifySettings(
+      WEX::TestExecution::VerifyOutputSettings::LogOnlyFailures);
+  LongVectorBinaryOpTestBase<float, 4>();
+}
+
+template <typename T, std::size_t N>
+//typename = std::enable_if_t<std::is_arithmetic_v<T>>>
+void ExecutionTest::LongVectorBinaryOpTestBase() {
+  WEX::TestExecution::SetVerifyOutput verifySettings(
+      WEX::TestExecution::VerifyOutputSettings::LogOnlyFailures);
+
+  // TODOLONGVEC: This seems frivolous as the device we target is created
```

Comment:
Will look at the rest of the test code and update this.
We may also want to consider a member variable based on whether or not this is running in the HLK vs as an exec test. Skipping should never be allowed in the HLK for any reason.